tooling/nargo/src/foreign_calls/default.rs

```diff
@@ -15,24 +15,51 @@ use super::rpc::RPCForeignCallExecutor;
 
 /// A builder for [DefaultForeignCallLayers] where we can enable fields based on feature flags,
 /// which is easier than providing different overrides for a `new` method.
-#[derive(Default)]
 pub struct DefaultForeignCallBuilder<'a> {
     pub output: PrintOutput<'a>,
+    pub enable_mocks: bool,
+
     #[cfg(feature = "rpc")]
     pub resolver_url: Option<String>,
+
     #[cfg(feature = "rpc")]
     pub root_path: Option<std::path::PathBuf>,
+
     #[cfg(feature = "rpc")]
     pub package_name: Option<String>,
 }
 
+impl<'a> Default for DefaultForeignCallBuilder<'a> {
+    fn default() -> Self {
+        Self {
+            output: PrintOutput::default(),
+            enable_mocks: true,
+
+            #[cfg(feature = "rpc")]
+            resolver_url: None,
+
+            #[cfg(feature = "rpc")]
+            root_path: None,
+
+            #[cfg(feature = "rpc")]
+            package_name: None,
+        }
+    }
+}
+
 impl<'a> DefaultForeignCallBuilder<'a> {
     /// Override the output.
     pub fn with_output(mut self, output: PrintOutput<'a>) -> Self {
         self.output = output;
         self
     }
 
+    /// Enable or disable mocks.
+    pub fn with_mocks(mut self, enabled: bool) -> Self {
```

Comment:
unused?

Reply:
It is, it looks like everywhere I wanted control over it I was already using the struct syntax, but because `with_output` exists, it looked like `with_mocks` is a nice to have as well for anyone who prefers the fluid syntax. It's a public type, so someone might use it I suppose. Or we could remove both, since the rest don't have them, although they could, because they go together I think, a method could nudge people to provide both.

Reply:
All good: just want to avoid accumulating unused code since `pub` interfaces can be tricky to remove.